ip_test.go

```diff
@@ -209,13 +224,13 @@ func TestPublicIPReservationFailFast(t *testing.T) {
 	// this should be an absurdly high number
 	quantity := 256
 
-	customData := map[string]interface{}{"custom1": "data", "custom2": map[string]interface{}{"nested": "data"}}
+	customData := `{"custom1":"data","custom2":{"nested":"data"}}`
 
 	req := IPReservationRequest{
 		Type:                   PublicIPv4,
 		Quantity:               quantity,
 		Facility:               &testFac,
-		CustomData:             customData,
```

Comment:
The `map[string]interface{}` tests should still work, do they not?  These would save JSON structures in the EM API, and would result in JSON structures in return.

The updated tests will be storing strings and I assume receiving strings.

Reply:
I thought we established that there can be non-map types coming from the EM API (strings, lists, numbers). Then, map[string]interface{} woulndn't work I think.

```
panic: interface conversion: interface {} is string, not map[string]interface {}
```

Reply:
@t0mk you can ignore that comment, I believe it is from before the investigation and discovery of how customdata is stored and delivered.

Reply:
Oh, I misunderstood myself :-)

I mean, we should still be able to test with: `customData := map[string]interface{}{"custom1": "data", "custom2": map[string]interface{}{"nested": "data"}}`. This would verify that arbitrary json structures are stored in their native format and returned the same.

By testing with: `{"custom1":"data","custom2":{"nested":"data"}}` we are verifying that scalars (string) work, and we could test with a simpler string `foo` to prove that.

Reply:
Both tests should be valid.

Reply:
Ah, you mean the concrete testing value in the Public IP reservation test. You're right that nested map[string]interface{} is more fitting here, thanks. I overlooked that there was still a string. I fixed it. 

I also changed the name of this PR.